packages/teleport/src/telemetry-boot.ts

```diff
@@ -0,0 +1,59 @@
+import { Resource } from '@opentelemetry/resources';
+import { SemanticResourceAttributes } from '@opentelemetry/semantic-conventions';
+import { WebTracerProvider } from '@opentelemetry/sdk-trace-web';
+import { registerInstrumentations } from '@opentelemetry/instrumentation';
+import {
+  ConsoleSpanExporter,
+  BatchSpanProcessor,
+} from '@opentelemetry/sdk-trace-base';
+import { OTLPTraceExporter } from '@opentelemetry/exporter-trace-otlp-http';
+import { DocumentLoadInstrumentation } from '@opentelemetry/instrumentation-document-load';
+import { XMLHttpRequestInstrumentation } from '@opentelemetry/instrumentation-xml-http-request';
+import { FetchInstrumentation } from '@opentelemetry/instrumentation-fetch';
+import { B3Propagator } from '@opentelemetry/propagator-b3';
+import { UserInteractionInstrumentation } from '@opentelemetry/instrumentation-user-interaction';
+import {
+  CompositePropagator,
+  W3CTraceContextPropagator,
+} from '@opentelemetry/core';
+import { ZoneContextManager } from '@opentelemetry/context-zone';
+
+export function instantiateTelemetry() {
+  registerInstrumentations({
+    instrumentations: [
+      new DocumentLoadInstrumentation(),
+      new UserInteractionInstrumentation(),
+      new XMLHttpRequestInstrumentation(),
+      new FetchInstrumentation(),
+    ],
+  });
+
+  const resource = Resource.default().merge(
+    new Resource({
+      [SemanticResourceAttributes.SERVICE_NAME]: 'web-ui',
+      [SemanticResourceAttributes.SERVICE_VERSION]: '0.1.0',
+    })
+  );
+
+  const provider = new WebTracerProvider({
+    resource: resource,
+  });
+
+  provider.addSpanProcessor(new BatchSpanProcessor(new ConsoleSpanExporter()));
+  provider.addSpanProcessor(
+    new BatchSpanProcessor(
+      new OTLPTraceExporter({
+        timeoutMillis: 15000,
+        url: `${window.location.origin}/v1/webapi/traces`,
```

Comment:
this doesn't work for me, i get 403 `need auth` error

Reply:
Right, sorry, I forgot about that. I believe that this will also not work if the UI is being served from webpack as webpack has not yet been configured to proxy the trace pings.